### ISSUE TITLE
Check doc warnings in `just build`

### ISSUE THREAD

**Issue by hannobraun** [hannobraun/Fornjot#773]
As of #766, documentation warnings are checked as part of the CI build. The same change hasn't been made to [`justfile`](https://github.com/hannobraun/Fornjot/blob/main/justfile) ([`CONTRIBUTING.md` has some context on what that file is for](https://github.com/hannobraun/Fornjot/blob/main/CONTRIBUTING.md#making-improvements)).

The check should be added there, to keep a local run of `just build` as close as reasonable to the CI build.

**Comment by jeevcat** [comment]
Should docs building warnings be treated as errors in this case too?

**Comment by hannobraun** [comment]
I just noticed an annoying side effect of setting the environment variables in `justfile`: Cargo won't use the existing artifacts from a previous variable-less build, meaning a `just build` after a regular build will trigger a full rebuild. But Cargo also won't put these new artifacts into a separate directory within `target/` (like it would when you build with a different `--target`), so the next regular build triggers a full rebuild again.

I wonder if there's a good way to work around this. Not failing and just showing warnings is probably not good enough, as the output from `just build` can get quite long, and it might be easy to miss warnings in there. Maybe it's possible to override Cargo's target directory for `just build`? Maybe to something like `target/just/`?

**Comment by jeevcat** [comment]
Okay if we don't use `RUSTFLAGS` it seems to not trigger a full rebuild. e.g. try this out:

```
export RUSTDOCFLAGS := "-D warnings"

build:
    cargo clippy --all-features -- -D warnings
    cargo test --all-features
    cargo fmt --check
    cargo doc --no-deps --document-private-items --all-features --workspace
    cargo run -p export-validator
```

**Comment by hannobraun** [comment]
Yes, but that could cause the problem I'm worried about: Warnings getting lost in the long output, without the build failing.

I just discovered `CARGO_TARGET_DIR`. Currently trying that out.

**Comment by jeevcat** [comment]
Sorry, I didn't mention it explicitly: I added `-- -D warnings` to the `clippy` command above. This will still fail the build.

**Comment by hannobraun** [comment]
> Sorry, I didn't mention it explicitly: I added `-- -D warnings` to the `clippy` command above. This will still fail the build.

Ah, got it. That seems fine then? At least I can't think of a problem.

---

The following is the result of the investigation I did since posting my previous comment here. It's probably moot now, but I figured I'd post it for the sake of completeness. Might come in handy in some way.

Setting `CARGO_TARGET_DIR` to `target/just/` has the intended effect, but it causes `export-validator` to fail because of this relative path:
https://github.com/hannobraun/Fornjot/blob/dec55059d59f420fe7b26a0ecfc9e70296c28ef0/tools/export-validator/build.rs#L18-L19

Maybe there's a cleverer way to build that path (using `CARGO_MANIFEST_DIR`?), or maybe we could set the output directory for `just build` to `target_just` instead of `target/just` (not my favorite, as that would further pollute the repository root directory).
